src/public/js/rs.js

```diff
@@ -123,45 +123,56 @@ var RS = (function() {
     var scaledGamma;
     var DThreshold;
     for (var i = 0; i <= n - k; i++) {
+      sigma = nextSigma;
+      omega = nextOmega;
+      tau = nextTau;
+      gamma = nextGamma;
+      D = nextD;
+      B = nextB;
+
       // Compute the left-hand side of the key equation
-      lhs = UTIL.polynomialMult(sigmas[i], syndromePolyPlusOne);
+      lhs = UTIL.polynomialMult(sigma, syndromePolyPlusOne);
 
       // Delta is the coefficient of z^(i+1)
       delta = lhs[(lhs.length - 1) - (i + 1)];
 
       // Use tau and gamma to update sigma and omega:
-      shiftedTau = UTIL.mergeTypedArrays(taus[i], new Uint8Array([0])); // z * tau
-      shiftedGamma = UTIL.mergeTypedArrays(gammas[i], new Uint8Array([0])); // z * gamma
+      shiftedTau = UTIL.mergeTypedArrays(tau, new Uint8Array([0])); // z * tau
+      shiftedGamma = UTIL.mergeTypedArrays(gamma, new Uint8Array([0])); // z * gamma
       scaledTau = UTIL.polynomialScale(shiftedTau, delta); // delta * z * tau
       scaledGamma = UTIL.polynomialScale(shiftedGamma, delta); // delta * z * gamma
-      sigmas[i + 1] = UTIL.polynomialAdd(sigmas[i], scaledTau);
-      omegas[i + 1] = UTIL.polynomialAdd(omegas[i], scaledGamma);
+      nextSigma = UTIL.polynomialAdd(sigma, scaledTau);
```

Comment:
Ah I see! You are correct, I missed the or equals too. My bad.

However this leads me to think that it could be done in 1 less iteration and nextSigma/nextOmega could be returned?